lib/web/terminal.go

```diff
@@ -299,6 +299,17 @@ func (t *TerminalHandler) streamTerminal(ws *websocket.Conn, tc *client.Teleport
 	// either an error occurs or it completes successfully.
 	err := tc.SSH(t.terminalContext, t.params.InteractiveCommand, false)
 
+	// Catches remote command exit errors to prevent sending a close event to UI,
+	// which prevents the UI terminal from closing the tab before user sees errors.
```

Comment:
This comment is incorrect as the purpose of this change is to stop swallowing errors during session initialization. In other words, if remote process exists with an error code, make sure we handle it.

Reply:
revised comment: https://github.com/gravitational/teleport/pull/4088/commits/a7564cca82d40bf719208c272ebe6a2d00aca73c

Reply:
Does it still creates a session record? (can you view it in the Audit Logs)? (it should not)
Does it show an actual error text? (Looks like it does not. If so, we may want to introduce our own "ExitCode" value for the errors we know (such as NotFound login) and handle it accordingly by mapping them to description text. 

Also, it might be actually **a feature** where we write errors into the teleport session stdout stream upon failing to create an SSH session :) If this is the case, lets try to be "smart" on the client where necessary.

Reply:
> Does it still create a session record? (can you view it in the Audit Logs)? (it should not)

Yes and yes. I was thinking maybe that this is a correct behavior because we want to log users who has a valid certificate to ssh into correct node, which starts a successful session, but b/c the user doesn't exist as a system user, exec command fails. I think this is what you meant by it being a feature?

> Does it show an actual error text? 

Yes, b/c at this point, websocket c/n has been established, and error are wrote back into it as displayed below:
![Screenshot from 2020-07-20 23-39-38](https://user-images.githubusercontent.com/43280172/88104948-08000100-cb58-11ea-9196-b107efd2963b.png)